src/app/events/InteractionCreate.ts

```diff
@@ -21,42 +21,28 @@ import DMPermissionError from '../errors/DMPermissionError';
 import ErrorUtils from '../utils/ErrorUtils';
 
 export default class implements DiscordEvent {
-    name = 'messageReactionAdd';
+    name = 'interactionCreate';
     once = false;
 
-    async execute(reaction: MessageReaction, user: User | PartialUser): Promise<any> {
-        // When a reaction is received, check if the structure is partial
-        if (reaction.partial) {
-            // Log.info('Pulling full reaction from partial');
-            await reaction.fetch();
-        }
-
-        if (user.partial) {
-            // Log.info('Pulling full user from partial');
-            try {
-                await user.fetch();
-            } catch (error) {
-                LogUtils.logError('failed to pull user partial', error);
-                return;
-            }
-        }
-
+    async execute(interaction: ButtonInteraction): Promise<any> {
+        const user: User = interaction.user;
+        if (!interaction.isButton()) return;
```

Comment:
What about existing bounty card messages that have reactions on them? We either need to convert them or still support emoji reactions.

I suggest supporting the reactions, but converting the cards to buttons when a user does a reaction. Will have a bunch of deprecated code, but not sure how else to do it.